File: schrottverwaltung/src/main/java/de/fhdortmund/schrottverwaltung/immobilie/AdresseT.java
package de.fhdortmund.schrottverwaltung.immobilie;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "adresse")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AdresseT {

    @Column(name = "strasse")
    private String strasse;

    @Column(name = "hausnummer")
    private int hausnummer;

    @Column(name = "hausnummer_zusatz")
    private String hausnummerZusatz;

    @Column(name = "plz")
    private int plz;

    @Column(name = "ort")
    private String ort;

    @Column(name = "stadtbezirk")
    private String stadtbezirk;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "adresse_id")
    private Long id;
}

File: schrottverwaltung/src/main/kotlin/de/fhdortmund/schrottverwaltung/immoblilie/AdresseT.kt
package de.fhdortmund.schrottverwaltung.immoblilie

import jakarta.persistence.*

@Entity
@Table(name = "adresse")
data class AdresseT(
    @Column(name = "strasse")
    var strasse: String,
    @Column(name = "hausnummer")
    var hausnummer: Int,
    @Column(name = "hausnummer_zusatz")
    var hausnummerZusatz: String?,
    @Column(name = "plz")
    var plz: Int,
    @Column(name = "ort")
    var ort: String,
    @Column(name = "stadtbezirk")
    var stadtbezirk: String,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "adresse_id")
    var id: Long? = null
)

